components/automate-cli/cmd/chef-automate/verify.go

```diff
@@ -204,126 +185,244 @@ func (v *verifySystemdCreateFlow) runVerifySystemdCreateCmd(cmd *cobra.Command,
+		)
+		if err != nil {
+			return err
+		}
+
+		c := NewVerifyCmdFlow(httputils.NewClient(log), createSystemdServiceWithBinary, config.NewHaDeployConfig(), sshutils.NewSSHUtil(sshutils.NewSshClient(), log), writer)
 		return c.runVerifyCmd(cmd, args, flagsObj)
 	}
 }
 
 func (v *verifyCmdFlow) runVerifyCmd(cmd *cobra.Command, args []string, flagsObj *verifyCmdFlags) error {
-	var configPath = ""
-	if len(args) > 0 {
-		configPath = args[0]
+	var configPath string
+
+	// TODO : config flag is optional for now. Need to handle the default config path
+	if len(strings.TrimSpace(flagsObj.config)) > 0 {
+		configPath = flagsObj.config
 	}
 
-	switch true {
-	case flagsObj.haAWSProvision:
-		err := v.verifyHaAWSProvision(configPath)
-		return err
-	case flagsObj.haAWSManagedProvision:
-		err := v.verifyHaAWSManagedProvision(configPath)
-		return err
-	case flagsObj.haAWSDeploy:
-		err := v.verifyHaAWSDeploy(configPath)
-		return err
-	case flagsObj.haAWSManagedDeploy:
-		err := v.verifyHaAWSManagedDeploy(configPath)
-		return err
-	case flagsObj.standaloneDeploy:
-		err := v.verifyStandaloneDeploy(configPath)
-		return err
-	case flagsObj.haOnpremDeploy:
-		err := v.verifyHaOnpremDeploy(configPath)
-		return err
-	case flagsObj.haOnPremAWSManagedDeploy:
-		err := v.verifyHaOnPremAWSManagedDeploy(configPath)
-		return err
-	case flagsObj.haOnPremCustManagedDeploy:
-		err := v.verifyHaOnPremCustManagedDeploy(configPath)
-		return err
-	case flagsObj.certificates:
-		err := v.Verification.VerifyCertificates("")
+	v.prettyPrint = flagsObj.prettyPrint
+
+	err := v.Config.ParseAndVerify(configPath)
+	if err != nil {
 		return err
-	default:
-		fmt.Println(cmd.UsageString())
-		return nil
 	}
 
-}
+	// Get config required for batch-check API call
+	batchCheckConfig := &models.Config{}
+	batchCheckConfig.PopulateWith(v.Config)
 
-func (v *verifyCmdFlow) verifyHaAWSProvision(configPath string) error {
-	if err := v.Verification.VerifyHAAWSProvision(configPath); err != nil {
-		return status.Annotate(err, status.ConfigError)
-	}
+	// TODO: For now just print the result as json. Need to merge the result with the response from batch-check API call for remote.
 
-	return nil
-}
+	err = v.checkAutomateVerifyServiceForBastion(*batchCheckConfig)
+	if err != nil {
+		return err
+	}
 
-func (v *verifyCmdFlow) verifyHaAWSManagedProvision(configPath string) error {
-	if err := v.Verification.VerifyHAAWSManagedProvision(configPath); err != nil {
-		return status.Annotate(err, status.ConfigError)
+	err = v.checkAutomateVerifyServiceForRemote(*batchCheckConfig)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func (v *verifyCmdFlow) verifyHaAWSDeploy(configPath string) error {
-	if !v.ManagedServicesOn {
-		if v.A2HARBFileExist {
-			if err := v.Verification.VerifyHAAWSDeployment(configPath); err != nil {
-				return status.Annotate(err, status.ConfigError)
+func (v *verifyCmdFlow) checkAutomateVerifyServiceForBastion(batchCheckConfig models.Config) error {
```

Comment:
break into multiple functions one for check and other for api call

Reply:
done